index.js

```diff
@@ -222,13 +222,16 @@ function fail(msg) {
 function showHistory(friendName) {
   var events = loadEventsData();
   var eventsByFriend = _.groupBy(events, "name");
-  Object.keys(eventsByFriend).sort().forEach(friend => {
-    if (!friendName || friendName === friend) {
-      prettyPrintFriendHeader(friend);
-      _.sortBy(eventsByFriend[friend], "date").reverse().map(prettyPrintEvent);
-      console.log();
-    }
-  });
+  var perFriend = function(friend) {
```

Comment:
I think this should be extracted into a top-level function, call it `showHistoryFor(friend)` or `showHistoryForFriend(friend)` perhaps.